templates/builders.html

```diff
@@ -0,0 +1,48 @@
+{% extends 'layout.html' %}
+{% import 'forms.html' as forms %}
+{% from "box_macros.html" import box %}
+
+{% block content %}
+<h1>Builders: {{ branches|join(', ')|e }}</h1>
+
+<table>
+{% for b in builders %}
+  <tr>
+  <td class="box"><a href="{{ b.link }}">{{ b.name|e }}</a></td>
+  {% if b.build_url %}
+    <td class="LastBuild box {{ b.build_css_class }}">
+        <a href="{{ b.build_url }}">{{ b.build_label }}</a>
+        <br/>{{ b.build_text }}
+    </td>
+  {% else %}
+    <td class="LastBuild box">no build</td>
+  {% endif %}
+  {{ box(**b.current_box) }}
+  </tr>
+{% endfor %}
+</table>
+
+{% if num_building > 0 %}
+  {% if authz.advertiseAction('stopAllBuilds', request) or authz.advertiseAction('stopBuild', request) %}
+    <h2>Stop Selected Builds</h2>
+    {{ forms.stop_build(path_to_root+"builders/_selected/stopselected", authz, on_selected=True, builders=builders, label='Selected Builds') }}
+    <h2>Stop All Builds</h2>
+    {{ forms.stop_build(path_to_root+"builders/_all/stopall", authz, on_all=True, label='All Builds') }}
+  {% endif %}
+{% endif %}
+
+{% if authz.advertiseAction('cancelAllPendingBuilds', request) %}
+  <h2>Cancel Selected Pending Builds</h2>
+  {{ forms.cancel_build(path_to_root+"builders/_selected/cancelpendingselected", authz, on_selected=True, builders=builders, label='Selected Pending Builds') }}
+  <h2>Cancel All Pending Builds</h2>
+  {{ forms.cancel_build(path_to_root+"builders/_all/cancelpendingall", authz, on_all=True, label='All Pending Builds') }}
+{% endif %}
+
+  {% if authz.advertiseAction('forceAllBuilds', request) or authz.advertiseAction('forceBuild', request) %}
+    <h2>Force Selected Builds</h2>
+    {{ forms.force_build(path_to_root+"builders/_selected/forceselected", authz, request, on_selected=True, builders=builders, force_schedulers=force_schedulers, default_props=default_props) }}
+    <h2>Force All Builds</h2>
+    {{ forms.force_build(path_to_root+"builders/_all/forceall", authz,request, on_all=True, force_schedulers=force_schedulers, default_props=default_props) }}
+  {% endif %}
+
+{% endblock %}
```

Comment:
I'm not familiar with this templating language so I'm just going to assume this works or if it doesn't it will be trivially and immediately obvious.